packages/rollup-plugin/src/index.ts

```diff
@@ -0,0 +1,83 @@
+import type { Plugin } from 'rollup';
+import {
+  cssFileFilter,
+  processVanillaFile,
+  compile,
+  IdentifierOption,
+  getSourceFromVirtualCssFile,
+  virtualCssFileFilter,
+} from '@vanilla-extract/integration';
+import { relative, normalize, dirname } from 'path';
+
+const virtualPrefix = 'virtual:vanilla-extract:';
+
+interface Options {
+  identifiers?: IdentifierOption;
+  cwd?: string;
+}
+export function vanillaExtractPlugin({ identifiers, cwd = process.cwd() }: Options = {}): Plugin {
+  const emittedFiles = new Map<string, string>();
+  const isProduction = process.env.NODE_ENV === 'production';
+
+  return {
+    name: 'vanilla-extract',
+    async transform(_code, id) {
+      if (!cssFileFilter.test(id)) {
+        return null;
+      }
+
+      const index = id.indexOf('?');
+      const filePath = index === -1 ? id : id.substring(0, index);
+
+      const { source, watchFiles } = await compile({
+        filePath,
+        cwd,
+      });
+
+      for (const file of watchFiles) {
+        this.addWatchFile(file);
+      }
+
+      return processVanillaFile({
+        source,
+        filePath,
+        identOption: identifiers ?? (isProduction ? 'short' : 'debug'),
+      });
+    },
+    async resolveId(id) {
+      // Only handle virtual css modules here
+      if (!virtualCssFileFilter.test(id)) {
+        return null;
+      }
+
+      // Emit the css asset
+      const { fileName, source } = await getSourceFromVirtualCssFile(id);
+      if (!emittedFiles.get(fileName)) {
+        const assetId = this.emitFile({
+          type: 'asset',
+          name: fileName,
+          source,
+        });
+        emittedFiles.set(fileName, assetId);
+      }
+      const assetId = emittedFiles.get(fileName);
+
+      // Resolve to a temporary virtual external chunk
+      return {
+        id: `${virtualPrefix}${assetId}`,
```

Comment:
Could we just return the `fileName` here seeing as it's external? That would mean we could remove all the `virtualPrefix` stuff and just use the `emittedFiles` Map for all the `renderChunk` logic.

Reply:
Done!